atomic_reactor/plugins/exit_remove_worker_metadata.py

```diff
@@ -0,0 +1,46 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from atomic_reactor.plugin import ExitPlugin
+from atomic_reactor.plugins.build_orchestrate_build import get_worker_build_info
+from atomic_reactor.constants import PLUGIN_REMOVE_WORKER_METADATA_KEY
+
+
+class RemoveWorkerMetadataPlugin(ExitPlugin):
+    """
+    Remove worker metadata for each platform.
+    """
+
+    key = PLUGIN_REMOVE_WORKER_METADATA_KEY
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+
+        # get all the build annotations and labels from the orchestrator
+        build_result = self.workflow.build_result
+
+        annotations = build_result.annotations
+        worker_builds = annotations['worker-builds']
+
+        for platform, build_annotations in worker_builds.items():
+            # retrieve all the workspace data
+            build_info = get_worker_build_info(self.workflow, platform)
+            osbs = build_info.osbs
+
+            kind = "configmap/"
+            cmlen = len(kind)
+            cm_key_tmp = build_annotations['metadata_fragment']
+
+            if not cm_key_tmp or cm_key_tmp[:cmlen] != kind:
+                self.log.warning("Bad ConfigMap %s for platform %s", cm_key_tmp, platform)
+                continue
+
+            cm_key = cm_key_tmp[cmlen:]
+            osbs.delete_config_map(cm_key)
+            self.log.debug("Deleting ConfigMap %s", cm_key)
```

Comment:
At this point you have already deleted, so "Deleting" is inaccurate.

Reply:
ok